pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -306,6 +306,7 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 	mounts, devices := util.GetPodVolumeNames(pod)
 
 	expandInUsePV := utilfeature.DefaultFeatureGate.Enabled(features.ExpandInUsePersistentVolumes)
+	volumeAdded := false
```

Comment:
to address to issue @msau42 mentioned. how about this logic
Initially volumeAdded := true
In the following for loop, if PodExistsInVolume returns false
volumeAdded = false
In this way, only if all volumes have been added to the desiredStateOfWorld before, volumeAdded is true.

Reply:
volumeAdded serves different purpose from that of the existing allVolumesAdded.
Along the above logic, I don't see why volumeAdded is needed - without which, the load on api server would still be at the high level.

Reply:
yes, they are different. 
1.allVolumesAdded will become false if dswp.createVolumeSpec and some others fail, which means it fails to process the volume for this round
2. volumeAdded becomes false if PodExistsInVolume returns false, which means that this volume has not been added to desired state before. 
As long as all volumes of this pod have been added to desired state once, volumeAdded should be true. Any subsequent processing the the pod, it does not matter whether allVolumesAdded is true or false, we treat as pod is processed.

Reply:
How about naming this variable allVolumesAddedPreviously ?
That way its meaning would be clearer.

Reply:
sure, that's good